common/transport/amqp/src/sender_link.ts

```diff
@@ -314,6 +315,22 @@ export class SenderLink extends EventEmitter implements AmqpLink {
             //
             let error = this._indicatedError; // This could be undefined.
             this._indicatedError = undefined;
+            if (!error) {
+              ///
+              // question for reviewers: Is this a Rhea bug?
+              //
+              // https://github.com/amqp/rhea/blob/e826e8bbffd410eaa5a1efc70650f37c14069b23/lib/link.js#L161-L163
+              //
+              // when they fire the link_error and link_close events, I think context.error should be set.
+              // Since it's not set, we have to guess that there's an error by looking at link state here.
+              //
+              // I think my change is OK?  Do you agree?  Should we push for a Rhea fix?
```

Comment:
This seems reasonable. Is there an issue filed against rhea for this? If so, we could keep an eye on it and remove this if they fix it